src/frontend/python/README.md

```diff
@@ -0,0 +1,41 @@
+# Quick Start
+## Setup
+Chiquito's Python frontend uses PyO3 and Maturin to expose Rust APIs to Python. Maturin requires the user to locally build a Python virtual environment. Run the following script to create a Python virtual environment, install required packages, and build the project.
+
+```
+# Clone this repo and its submodules
+git clone --recursive https://github.com/privacy-scaling-explorations/chiquito/
```

Comment:
I think is from the time pychiquito, this script has to be checked in a clean environment inside a docker for example

Reply:
@qwang98 did you see this?

Reply:
I tried the scripts in a Docker using the .devcontainer file that's in main (which I think you pushed).

Everything worked fine up through `maturin develop`, for which I received a low level Linker error which I couldn't debug:

```
error: linking with `cc` failed: exit status: 1
  |
  = note: LC_ALL="C" PATH="/usr/local/rustup/toolchains/nightly-2023-04-24-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/bin:/workspaces/chiquito/.env/bin:/vscode/vscode-server/bin/linux-arm64/660393deaaa6d1996740ff4880f1bad43768c814/bin/remote-cli:/home/vscode/.local/bin:/usr/local/python/current/bin:/usr/local/py-utils/bin:/usr/local/cargo/bin:/usr/local/cargo/bin:/usr/local/python/current/bin:/usr/local/py-utils/bin:/usr/local/cargo/bin:/usr/local/cargo/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/home/vscode/.local/bin" VSLANG="1033" "cc" "-Wl,--version-script=/tmp/rustcElE4Wg/list" "-Wl,--no-undefined-version" "/tmp/rustcElE4Wg/symbols.o" "/workspaces/chiquito/target/debug/deps/chiquito.13argb33nwbba5xi.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.13evmipldvpnd2m7.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.13xew0txjbvtqn0g.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.14e1pcfklfkx6snp.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.17l0c8tp35beyao9.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.17z0x0af8pjf1wj9.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.18pbt5b009d00bl7.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.1amu7nh5qsv318wz.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.1aqswmdktbqpae7k.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.1ay0gzrsw42p4pzy.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.1bj2ig05jwxmb22h.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.1c1kamp9umik4e77.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.1cvs9jcxpghq8omy.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.1d1i0sa2a3zmoho9.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.1em29hgi4x5b2vj9.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.1gqhh12i8zbjybzt.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.1ic82wkaeyvz20ic.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.1kajuo7b86ggj0wc.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.1ms5aqhls0rodxbh.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.1ohv62ew3ykdchdz.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.1oza5ud7undvhu6h.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.1r0gcq325ufhnj18.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.1r2032ydvrutv9y1.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.1rk7bjsk1dm4wd12.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.1sjlzviagwa0d6me.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.1srvkptmhnr1syj4.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.1te4l2aseftsicqy.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.1uw9hdwkvv10l76i.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.1vm5ko85ejxe0sgl.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.1w9x406ejtdv67v.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.1zcsc6jmakcoh2l3.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.1zmz2utdi9rhd66c.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.1zqofi1w67dru6rv.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.20niodn325pdkcfz.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.21jjxo6pvnrujqdg.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.21t3gponobpj6zmj.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.22eekysxmlwigweb.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.22op9k4o11y3wdtb.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.253dpehlwh1lp5d9.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.25hnynl2uigg768m.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.26qx0of4pzmnpbmn.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.27hh2y7qm8sctpip.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.28l0atleog16v8z9.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.29evyfnay5ytpjnv.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.2a55u2jbmdjd3w8q.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.2bnrtq2rtudhvbjj.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.2dein97o1skq4602.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.2dibi53s4e3o658p.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.2g1nmaa9fou0fe31.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.2gz18q61tw9bj55a.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.2hzcr22g2t33pm9c.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.2kh6idwiayz0c5j8.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.2kiv70hkb51pktqc.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.2kpqct6nguc05w7m.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.2lgwo94a64d76yg.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.2m2o5sdc6bxy0hku.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.2msrzjhdppcpt7wh.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.2n3xncxhbn4pi60t.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.2nafi2mw8fxtmuev.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.2qyynenu2mos6igg.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.2r9q7oqsyonxqn4j.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.2rgmpq6m3uma7yg8.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.2rh6xhqws5tsq505.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.2rsu7otlrjqwp4am.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.2sx3cwxzeca4141i.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.2tf9vs3kuz2ir1gh.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.2v7g70nwfnhyt2u0.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.2x3x8zfcypvate3c.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.2ysmzdbdngk0eekn.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.2yt4s74m6ziqolpu.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.2yzsicw6j33ekl97.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.2zftapwpmwhzsxdf.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.2zunoj4bjj48oblz.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.30ywtl5rnasbcld5.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.32cs60mhx0lfpwqi.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.32rda6mayr6xdulu.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.333tmngd6jorob6l.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.34x3dy3fv39x551b.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.36e3pvoe5bd1p3kn.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.36mehs9ni41dufpo.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.39an8gj3j6uvjsut.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.39vafanki4706hv1.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3a894hjjiz83alo2.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3bf2idyfa1kyc6sm.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3bjclrafnt32ngpy.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3cw496k9wsx329u3.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3d79jqzz857xsajn.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3dn9tkm5ct2zuga7.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3ebu3h48gs0qep5c.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3elwwjsdw9irseuv.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3ffikc27cqbw0zva.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3fhywyxhvfukssr8.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3fpoczdhsqe88340.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3fuz0aald1rku7jc.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3gr9pot0ze5qyh6a.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3j7571k1tr4bdw4l.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3jc568hwwxt4ye93.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3k2nm6gyvappfxkf.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3kjuitmrpu2xro3i.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3kwqnbbw5i1dqgwc.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3l25h54w0m6qaaxu.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3l3e7dbd7cjsw4p2.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3m2ly77mpdubbk91.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3nl8lcpopcq2j13t.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3pd5vvk5d1ikg1nz.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3q3oomfn4j3n6tvd.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3q74mbfiasqioq7m.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3qaw417kl63we88p.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3qscpc5ov937zui7.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3rlbnz1wtax1btmb.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3sj8m8cbk6y02qrs.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3sjbhd1nuc4xidy.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3tbt76mi9qqhc0up.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3tx6kw45n70nknw3.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3w4ziwhp26ggfsmw.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3w56o33yf8f65ut.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3x3cm8dyfrv8ects.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3x3t221179dtay1c.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3x4pa9ys9ratyczz.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3x4z4uj27g1iepkr.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3xz4r1nf8cphfuuc.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3yrz7s1al0m116oa.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3yxz5itokqn04rn5.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.3z7kmnr4lah1muh.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.427nmjhth8hblysu.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.42ehrxlz2vpvs69d.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.42oat0cgi1idmzbj.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.43cpwt49s5iqyfqs.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.43yoi6f8gdgfu64z.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.44phpltk2y2wopr4.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.44suhrld5r2j6ghf.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.44yo4ju5xplurd5j.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.451lx9k0mjjmqfy8.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.458jx60l5081bznt.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.486wcpx2lkkzwdbq.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.499o1qawcrau88mh.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.49bniaw2yp5fzw5h.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.4h254pdm3fjpmixz.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.4h7kbtyz1nn4jw85.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.4hjqtg89e27fmqq3.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.4iemak09rswwt79k.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.4jbig9znft6cow77.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.4jeik8xj90vlmodg.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.4jokflubs88uj8is.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.4jpnzpcjnd4v46lo.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.4k6mndu0f4o78rjg.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.4kdx09kk8j39j2yu.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.4mi9rblsivtvfl8h.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.4ngl75tpxyvb1qcw.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.4pjp9ae2aqj4ujov.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.4pkv58rgnwj2vxzg.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.4qmvu8vwmm7yxmj1.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.4r52rb3uzpva4y2t.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.4rf74i90aacozjz1.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.4sov00290nzy5lnt.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.4w8f0c2cbvogxgjt.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.4xk7046i48qz6njd.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.4xv425p1ffyotv5q.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.4xzyy4w6t0xkkpf1.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.4y73i8q87gq17xgh.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.4yyyt9c1u6qlp3up.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.4z9um139d71lrbrb.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.51ywt0f92c3ypxt0.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.52s3a570ys93gp2s.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.558uxphnjk6au110.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.55zrxdnvs82sjx8x.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.561q7oqrm65z69hy.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.56glrv9so0tcz8jb.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.57hdxfzedvevhtni.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.586f9wtjdxuxzlrw.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.59g4p4cmk3xwqhx8.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.59qk91n0m21qska5.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.5bcbu4jfimw1gced.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.5dvksx76kxm036e4.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.5ecwq75v7dwa6s2p.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.5fvk9eqy1omveway.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.690evbfh5w1w1l5.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.71kpnylksmv0oas.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.75l5d6eyzhgu2pq.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.8ciw90wowt7r9dg.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.a7qefyaabcxm1ua.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.bxt1i1nejsr8fg6.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.cwv7txem4xj3gfx.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.d8sjdb2ottm645i.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.ejj80huccywc97l.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.evb599ivj85eci3.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.fm4ac1bi2m33zmd.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.g6ega6iye0b9l8r.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.gfqlyw0tiykwtmv.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.h4d49kjqg414l7j.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.hmwuejhjktq990t.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.hxyznomxwb1jo0i.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.j09dlx9bps56vg3.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.jt1fmgni9zdbo4v.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.kmd1so4zy3rw3sn.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.la524qan0z1z3e3.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.lewyalsbw04s3qo.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.lvx83pwz9cdjpvc.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.mijyw3922k6ndpq.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.njlkbbtujdelzyz.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.noi8lfcfsrs8eyo.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.nw9bnailvpilhla.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.pnoda84p1e5wwtu.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.qhy8z33atthcajv.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.qy2hfrusaqy8lv8.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.r3ljscytkjn6nty.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.wsmp15113zybc5h.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.x61utgr76956pk7.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.xsox8l0u8q9hew5.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.y3lc6sjnndipdq0.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.yf8npl4ujnlabty.rcgu.o" "/workspaces/chiquito/target/debug/deps/chiquito.44yh6wmymqgefo0h.rcgu.o" "-Wl,--as-needed" "-L" "/workspaces/chiquito/target/debug/deps" "-L" "/usr/local/rustup/toolchains/nightly-2023-04-24-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib" "-Wl,-Bstatic" "/workspaces/chiquito/target/debug/deps/libserde_json-7085dc704f9dfd15.rlib" "/workspaces/chiquito/target/debug/deps/libryu-17241e95fc31445d.rlib" "/workspaces/chiquito/target/debug/deps/libitoa-ab3b47809a9a5b9f.rlib" "/workspaces/chiquito/target/debug/deps/libuuid-fbf3d96da78cc282.rlib" "/workspaces/chiquito/target/debug/deps/libatomic-1b84aa287c94137d.rlib" "/workspaces/chiquito/target/debug/deps/libpolyexen-66d2ebb976fbb8bb.rlib" "/workspaces/chiquito/target/debug/deps/libpest-c750d98ea4d832af.rlib" "/workspaces/chiquito/target/debug/deps/libucd_trie-089b62f761595f4f.rlib" "/workspaces/chiquito/target/debug/deps/libthiserror-95cb812ea5828d86.rlib" "/workspaces/chiquito/target/debug/deps/libnum_bigint-b5ff5214f7188cee.rlib" "/workspaces/chiquito/target/debug/deps/libnum_integer-79346a36a7d37f12.rlib" "/workspaces/chiquito/target/debug/deps/libnum_traits-ec168b632a2f9197.rlib" "/workspaces/chiquito/target/debug/deps/libpyo3-20668790fe0964e9.rlib" "/workspaces/chiquito/target/debug/deps/libparking_lot-57916408f6d0816d.rlib" "/workspaces/chiquito/target/debug/deps/libparking_lot_core-713772e64959d8e5.rlib" "/workspaces/chiquito/target/debug/deps/libsmallvec-29804136eebf80a4.rlib" "/workspaces/chiquito/target/debug/deps/liblock_api-feeaba9d13788dc9.rlib" "/workspaces/chiquito/target/debug/deps/libpyo3_ffi-a8c0390df97c2434.rlib" "/workspaces/chiquito/target/debug/deps/libunindent-cbfed5337c257805.rlib" "/workspaces/chiquito/target/debug/deps/libhalo2_proofs-b3dc321a7d955f18.rlib" "/workspaces/chiquito/target/debug/deps/libtracing-5fea7a1d6e4608df.rlib" "/workspaces/chiquito/target/debug/deps/libpin_project_lite-3c37242582fa112c.rlib" "/workspaces/chiquito/target/debug/deps/libtracing_core-f79229a79ee905a4.rlib" "/workspaces/chiquito/target/debug/deps/libonce_cell-01c8eb977c0ba4a4.rlib" "/workspaces/chiquito/target/debug/deps/libsha3-54e3678ae4ee6ad7.rlib" "/workspaces/chiquito/target/debug/deps/libkeccak-2bab1d6d816d9c7b.rlib" "/workspaces/chiquito/target/debug/deps/libopaque_debug-97a16812bdc0c27c.rlib" "/workspaces/chiquito/target/debug/deps/libblock_buffer-244f70cf2445835e.rlib" "/workspaces/chiquito/target/debug/deps/libblock_padding-28cddfaae3cb6a4a.rlib" "/workspaces/chiquito/target/debug/deps/libdigest-b726ef18d3e2b82a.rlib" "/workspaces/chiquito/target/debug/deps/libgeneric_array-bc5516ebfbc9c8b7.rlib" "/workspaces/chiquito/target/debug/deps/libtypenum-6b3140651e8239bf.rlib" "/workspaces/chiquito/target/debug/deps/librayon-f1c33152eae98b4f.rlib" "/workspaces/chiquito/target/debug/deps/librayon_core-439db9935429ae10.rlib" "/workspaces/chiquito/target/debug/deps/libnum_cpus-7c8c9502a5a500b2.rlib" "/workspaces/chiquito/target/debug/deps/libcrossbeam_deque-901aa1d660e5fed6.rlib" "/workspaces/chiquito/target/debug/deps/libcrossbeam_epoch-6513ecdad021e1a0.rlib" "/workspaces/chiquito/target/debug/deps/libmemoffset-e748ea21b0d77b7b.rlib" "/workspaces/chiquito/target/debug/deps/libscopeguard-b0999f1bfd4709e0.rlib" "/workspaces/chiquito/target/debug/deps/libcrossbeam_channel-ea8d4ab8ceac9a16.rlib" "/workspaces/chiquito/target/debug/deps/libcrossbeam_utils-0ee9189973cd72f6.rlib" "/workspaces/chiquito/target/debug/deps/libeither-dd32d57870151521.rlib" "/workspaces/chiquito/target/debug/deps/libhalo2curves-c5cb19077b3b999d.rlib" "/workspaces/chiquito/target/debug/deps/libpasta_curves-b72293e04e9439d1.rlib" "/workspaces/chiquito/target/debug/deps/libblake2b_simd-5535743d08e63094.rlib" "/workspaces/chiquito/target/debug/deps/libconstant_time_eq-dd87380ffae24523.rlib" "/workspaces/chiquito/target/debug/deps/libarrayvec-a73c8c7989535726.rlib" "/workspaces/chiquito/target/debug/deps/libarrayref-c380fb85248d98e5.rlib" "/workspaces/chiquito/target/debug/deps/libstatic_assertions-8111166027bb520a.rlib" "/workspaces/chiquito/target/debug/deps/liblazy_static-ba5c4d03d164c2fb.rlib" "/workspaces/chiquito/target/debug/deps/libspin-b1a7ce681b86ddb0.rlib" "/workspaces/chiquito/target/debug/deps/libserde-af785e79511a8f99.rlib" "/workspaces/chiquito/target/debug/deps/librand-57ec5033e5c881de.rlib" "/workspaces/chiquito/target/debug/deps/librand_chacha-cbef42df19e39921.rlib" "/workspaces/chiquito/target/debug/deps/libppv_lite86-a72204c13630a37a.rlib" "/workspaces/chiquito/target/debug/deps/libgroup-8bfd0894ee4036e9.rlib" "/workspaces/chiquito/target/debug/deps/libff-d40ec9f98764acdf.rlib" "/workspaces/chiquito/target/debug/deps/librand_core-5f85ae2057c3fa92.rlib" "/workspaces/chiquito/target/debug/deps/libgetrandom-38f9042a2ac4bee8.rlib" "/workspaces/chiquito/target/debug/deps/liblibc-90149872e5251b3e.rlib" "/workspaces/chiquito/target/debug/deps/libcfg_if-9761cf86ce87d124.rlib" "/workspaces/chiquito/target/debug/deps/libbitvec-28e3103482218ff9.rlib" "/workspaces/chiquito/target/debug/deps/libwyz-8bd7d65d1f06efe4.rlib" "/workspaces/chiquito/target/debug/deps/libtap-9750d07933851cee.rlib" "/workspaces/chiquito/target/debug/deps/libradium-d439b6952865eeb8.rlib" "/workspaces/chiquito/target/debug/deps/libfunty-db7578df0415c8e0.rlib" "/workspaces/chiquito/target/debug/deps/libsubtle-c46e98eebf94c98c.rlib" "/usr/local/rustup/toolchains/nightly-2023-04-24-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libstd-89cee6c3f7a8a248.rlib" "/usr/local/rustup/toolchains/nightly-2023-04-24-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libpanic_unwind-89f0b3a1354cbb39.rlib" "/usr/local/rustup/toolchains/nightly-2023-04-24-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libobject-5258c6a6f53ed405.rlib" "/usr/local/rustup/toolchains/nightly-2023-04-24-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libmemchr-dfc39cdcdc6fb39b.rlib" "/usr/local/rustup/toolchains/nightly-2023-04-24-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libaddr2line-6cfcfd042676bdfd.rlib" "/usr/local/rustup/toolchains/nightly-2023-04-24-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libgimli-a4cad6bd6a1ea319.rlib" "/usr/local/rustup/toolchains/nightly-2023-04-24-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/librustc_demangle-cd33bd97ca044183.rlib" "/usr/local/rustup/toolchains/nightly-2023-04-24-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libstd_detect-6696fffc6ceffbda.rlib" "/usr/local/rustup/toolchains/nightly-2023-04-24-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libhashbrown-ec16bc37044f2d5d.rlib" "/usr/local/rustup/toolchains/nightly-2023-04-24-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/librustc_std_workspace_alloc-55151f99f732c891.rlib" "/usr/local/rustup/toolchains/nightly-2023-04-24-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libminiz_oxide-a33eb3a46afc3767.rlib" "/usr/local/rustup/toolchains/nightly-2023-04-24-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libadler-6f5c6b6d790e5553.rlib" "/usr/local/rustup/toolchains/nightly-2023-04-24-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libunwind-89823c8400f5ecc3.rlib" "/usr/local/rustup/toolchains/nightly-2023-04-24-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libcfg_if-16ba045d4ec2c438.rlib" "/usr/local/rustup/toolchains/nightly-2023-04-24-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/liblibc-d2d232c6d4fe71c3.rlib" "/usr/local/rustup/toolchains/nightly-2023-04-24-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/liballoc-f5dff897100c1d47.rlib" "/usr/local/rustup/toolchains/nightly-2023-04-24-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/librustc_std_workspace_core-18ea8bd63a028a62.rlib" "/usr/local/rustup/toolchains/nightly-2023-04-24-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libcore-d20cd64dc93f727e.rlib" "/usr/local/rustup/toolchains/nightly-2023-04-24-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libcompiler_builtins-b142ce8e62e662c1.rlib" "-Wl,-Bdynamic" "-lgcc_s" "-lutil" "-lrt" "-lpthread" "-lm" "-ldl" "-lc" "-Wl,--eh-frame-hdr" "-Wl,-z,noexecstack" "-L" "/usr/local/rustup/toolchains/nightly-2023-04-24-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib" "-o" "/workspaces/chiquito/target/debug/deps/libchiquito.so" "-Wl,--gc-sections" "-shared" "-Wl,-z,relro,-z,now" "-nodefaultlibs"
  = note: /usr/bin/ld: cannot find /workspaces/chiquito/target/debug/deps/chiquito.la524qan0z1z3e3.rcgu.o: No such file or directory
          collect2: error: ld returned 1 exit status
```

Reply:
Everything worked just fine (including `maturin develop`) in my local machine, even after I deleted and reinstalled the virtual environment.

Reply:
I deleted reference to "submodules".

Reply:
I tested this in a dev container using the `.devcontainer` config folder from `main` (which I think you pushed).

For some reason, we need to configure chiquito version as `0.1.0` in Cargo.toml for `maturin develop` to not fail. Otherwise, I was able to successfully test the scripts in this README.md file in a clean container.

Reply:
`--recursive` is for submodules